### PR TITLE
feat(verifier): add Halo2 KZG Verifier

### PR DESCRIPTION
**Pull request by PatStiles**
Integrates halo2-kzg proof verification on aligned layer operators. New pr for #167 due to accidental submodule commit.

### CODE REVIEW

**Review by jotabulacios — APPROVED** [review]
lgtm. all tests passed successfully

File: Makefile [hunk 1]
```diff
@@ -208,6 +208,23 @@ batcher_send_halo2_ipa_task_burst_5: batcher/client/target/release/batcher-clien
...
 		--repetitions 5
 
+batcher_send_halo2_kzg_task: batcher/client/target/release/batcher-client
+	@echo "Sending Halo2 KZG 1!=0 task to Batcher..."
+	@cd batcher/client/ && cargo run --release -- \
+		--proving_system Halo2KZG \
+		--proof test_files/halo2_kzg/proof.bin \
+		--public_input test_files/halo2_kzg/pub_input.bin \
+		--vk test_files/halo2_kzg/params.bin \
+
```
Comment (NicolasRampoldi) [inline Makefile:219]: Missing `proof_generator_addr` flag

File: Makefile [hunk 2]
```diff
@@ -208,6 +208,23 @@ batcher_send_halo2_ipa_task_burst_5: batcher/client/target/release/batcher-clien
...
+batcher_send_halo2_kzg_task_burst_5: batcher/client/target/release/batcher-client
+	@echo "Sending Halo2 KZG 1!=0 task to Batcher..."
+	@cd batcher/client/ && cargo run --release -- \
+		--proving_system Halo2KZG \
+		--proof test_files/halo2_kzg/proof.bin \
+		--public_input test_files/halo2_kzg/pub_input.bin \
+		--vk test_files/halo2_kzg/params.bin \
+		--repetitions 5
+
```
Comment (NicolasRampoldi) [inline Makefile:229]: Missing `proof_generator_addr` flag

File: Makefile (hunk 1 above)
Comment (NicolasRampoldi) [reply Makefile:219]: Solved in [dc46037](https://github.com/yetanotherco/aligned_layer/pull/268/commits/dc460377b873c514d3fafe2f9009e4617ec3b09d)

File: Makefile (hunk 2 above)
Comment (NicolasRampoldi) [reply Makefile:229]: Solved in [dc46037](https://github.com/yetanotherco/aligned_layer/pull/268/commits/dc460377b873c514d3fafe2f9009e4617ec3b09d)

**Review by NicolasRampoldi — APPROVED** [review]

**Pull request merged**